extensions/podman/packages/extension/src/extension.ts

```diff
@@ -812,11 +813,13 @@ export async function registerProviderFor(
     };
   }
 
+  const providerConnectionShellAccess = new ProviderConnectionShellAccessImpl(machineInfo);
```

Comment:
here `providerConnectionShellAccess` should be (extends) a disposable and push to subscriptions